examples/rock-paper-scissors/game.rs

```diff
@@ -0,0 +1,494 @@
+use super::require;
+use core::cmp::Ordering;
+use ink_env::{
+    hash::{
+        HashOutput,
+        Sha2x256,
+    },
+    AccountId,
+    Hash,
+};
+use ink_primitives::KeyPtr;
+use ink_storage::traits::{
+    PackedLayout,
+    SpreadAllocate,
+    SpreadLayout,
+};
+use scale::{
+    Decode,
+    Encode,
+};
+
+type BlockNo = u32;
+
+/// How many blocks may pass until the game is considered to be timed out.
+pub(crate) const TIMEOUT: BlockNo = 100;
```

Comment:
Please use `BlockNumber` here, see e.g. https://github.com/paritytech/ink/blob/master/examples/mother/lib.rs#L72. You'll have to move those definitions inside the `#[ink::contract]` macro for it to work.

The context here is that the block number is a generic type in Substrate. So could be anything. By using ink!'s `BlockNumber` you'll use the chain's type.